code/create_survey_geojson.py

```diff
@@ -0,0 +1,150 @@
+    }
+
+    # split into states and clusters
+    survey_states = [
+        state for state in survey_states_and_clusters if "Cluster" in state
+    ]
+    survey_clusters = [
+        state
+        for state in survey_states_and_clusters
+        if state not in survey_states
+    ]
+
+    survey_state_info = []
+
+    # add single states
+    for state in survey_clusters:
+        print(f"Adding state {state}")
+        state_info = state_info_map[state]
+        state_info["id"] = state
+        del state_info["properties"]["density"]
+        survey_state_info.append(state_info)
+
+    # add clusters
+    for cluster in survey_states:
+
+        # get individual state names
+        try:
+            clustered_states = (
+                re.search("(.*) \\(Cluster", cluster).groups()[0].split(", ")
+            )
+        except Exception as exception:
+            raise RuntimeError(
+                f"Could not parse states in cluster {cluster}: {exception}"
+            )
+
+        print(f"Adding cluster {cluster}")
+        cluster_info = None
+        for clustered_state in clustered_states:
+            print(f"\tAdding state {clustered_state}")
+
+            # first state in the cluster
+            if cluster_info is None:
+                cluster_info = state_info_map[clustered_state]
+                cluster_info["properties"]["name"] = cluster
+                cluster_info["id"] = cluster
+
+                # convert into MultiPolygon if needed
+                if cluster_info["geometry"]["type"] == "Polygon":
+                    cluster_info["geometry"]["type"] = "MultiPolygon"
+                    cluster_info["geometry"]["coordinates"] = [
+                        cluster_info["geometry"]["coordinates"]
+                    ]
+
+            # subsequent states in the cluster
+            else:
+                state_info = state_info_map[clustered_state]
+                if state_info["geometry"]["type"] == "Polygon":
+                    to_extend = [
+                        state_info_map[clustered_state]["geometry"][
+                            "coordinates"
+                        ]
+                    ]
+                else:
+                    to_extend = state_info_map[clustered_state]["geometry"][
+                        "coordinates"
+                    ]
+                cluster_info["geometry"]["coordinates"].extend(to_extend)
+
+        del cluster_info["properties"]["density"]
+        survey_state_info.append(cluster_info)
+
+    survey_json = {
+        "type": "FeatureCollection",
+        "features": survey_state_info,
+    }
+    fpath_out.write_text(json.dumps(survey_json, indent=4))
+    print(f"GeoJSON file written to {fpath_out}")
+
+    return survey_json
+
+
+if __name__ == "__main__":
+
+    # TODO switch to data_loader SURVEY_RESULTS when that is merged
+    fpath_df_states = DPATH_DATA / "survey_results" / "sampledesc_state.tsv"
+    df_states = pd.read_csv(fpath_df_states, sep="\t")
```

Comment:
probably doable now, though it would create a dependency on the code that's in the app folder

so not even sure we should worry about now

I would wait until there is an actual need before refactoring

Reply:
For this script specifically, I think either is fine, since the script won't be called by the app itself.

The variables in `data_loader.py` are more relevant for plotting functions that will be directly used in the app, to cut down on time for file I/O